cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -192,6 +196,67 @@ public String toString() {
     }
   }
 
+  /**
+   * Represents a program status trigger for REST requests/responses
+   */
+  public static class ProgramStatusTrigger extends ProtoTrigger {
+    protected final ProgramId programId;
+    protected final Set<ProgramStatus> programStatus;
+
+    public ProgramStatusTrigger(ProgramId programId, Set<ProgramStatus> programStatus) {
```

Comment:
Does this ProgramId represent the Triggering program or the Triggered program?
My assessment is that it is the Triggering program. If that is the case, then the exception message below needs to be updated accordingly.